.github/workflows/test.yaml

```diff
@@ -16,3 +20,34 @@ jobs:
       - uses: actions/checkout@v2
       - name: Run ${{ matrix.command }}
         run: make ${{ matrix.command }}
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: ${{ env.COMPONENT_NAME }}
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/cache@v1
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Compile component
+        run: |
+          GO111MODULE="on" go get github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb
+          cd ${{ env.COMPONENT_NAME }}
+          make compile
+      - name: Determining tests to run
+        id: determine-tests
+        run: |
+          if [ -f "${{ env.COMPONENT_NAME }}/tests/go/go.mod" ]; then echo "::set-output name=RUN_GO_TESTS::true" ; fi
+          if [ -d "${{ env.COMPONENT_NAME }}/tests/conftest" ]; then echo "::set-output name=RUN_CONFTESTS::true" ; fi
+      - name: Run Unit tests
```

Comment:
My argument to have two separate test targets would be to run them in parallel. Here there they are run in sequence so we could just go with one target.

Reply:
The go tests and conftest run faster than the compile step. Parallelization at the moment makes it actually longer due to overhead of the jobs themselves.